### PR TITLE
Implement ATB active mode

### PR DESCRIPTION
**Pull request by rueter37**
This PR implements the ATB active mode. This fixes bullet point 3 of #2175.

### CODE REVIEW

File: src/scene_battle_rpg2k3.cpp [hunk 1]
```diff
@@ -261,29 +261,39 @@ void Scene_Battle_Rpg2k3::Update() {
 
 	switch (state) {
 		case State_SelectActor:
-		case State_AutoBattle: {
+		case State_AutoBattle:
+		case State_SelectCommand:
+		case State_SelectEnemyTarget:
+		case State_SelectAllyTarget: {
+			if (Game_System::GetAtbMode() == lcf::rpg::SaveSystem::AtbMode_atb_wait && (state == State_SelectCommand || state == State_SelectEnemyTarget || state == State_SelectAllyTarget)) break;
+			if ((state == State_SelectEnemyTarget || state == State_SelectAllyTarget) && (previous_state == State_SelectSkill || previous_state == State_SelectItem)) break;
```
Comment (Ghabry) [inline src/scene_battle_rpg2k3.cpp:269]: You check "case" values in an if and then break out of the switch.
I suggest putting them before ``State selectactor`` and then just ``// fallthrough``

Comment (rueter37) [reply src/scene_battle_rpg2k3.cpp:269]: I'm probably too stupid but how I'm supposed to do the fallthrough?

Comment (Ghabry) [reply src/scene_battle_rpg2k3.cpp:269]: To fallthrough just don't use break. It is just convention (compilers warn about it) that you must use ``// fallthrough`` (a comment) when the fallthrough was intended.

Something like this.

```
case State_SelectEnemyTarget:
case State_SelectAllyTarget:
  if (previous_state == State_SelectSkill || previous_state == State_SelectItem) {
    break;
  }
  // fallthrough
case State_SelectCommand:
  if (Game_System::GetAtbMode() == lcf::rpg::SaveSystem::AtbMode_atb_wait) {
    break;
  }
  // fallthrough
case State_SelectActor:
case State_AutoBattle:
 /* more code */
```

Comment (rueter37) [reply src/scene_battle_rpg2k3.cpp:277]: Thank you for your input! It helps me understanding the switch-case-operation a bit better. I have updated the PR.

**Review by Ghabry — APPROVED** [review]
Imo this is an anti feature but works for me.
Tested attacking while being killed through a skill, worked without crashing.
For later commits there shouldbe a checklist what RPG_RT does at these edge cases:

- Killed while starting normal attack
- Killed while opening Skill Window
- Game Over while opening Item window
etc.

**Review by fdelapena — APPROVED** [review]

**Review by fmatthew5876 — APPROVED** [review]
Code looks ok, if @Ghabry tested this one then it's good enough for me.

**Pull request merged**